### DISCUSSION TITLE
support Armachat

### DISCUSSION THREAD

**Discussion by srem1** [markqvist/Reticulum#215] [Ideas]
I like this dev https://hackaday.io/project/171790-armawatch-armachat-long-range-radio-messengers but I'm not have any ;)

Is possible create similar dev or talking with author armachat (and many other communicators) and implement this in rnode?

**Comment by markqvist** [comment]
Making devices like that, but done properly and built on Reticulum, has been one of my personal goals and interests from when i started this project.

There's a thousand variations of this same basic idea out there. Many of them are cool tech demos and fun projects, but ultimately they fall short of being truly good tools for real world use, since they are just monolithic applications, and nothing can interoperate with them. They focus on creating the cool and shiny parts *first*, and then building the *absolute minimum* amount of actual technological foundation for the shiny parts.

I do things the other way around. It takes a lot longer, but once the foundation is there, you can build anything on top, and it tends to work a lot better. 

It *will* come, but I am interested in building platforms and tools, not (just) toys. That means there is a few more puzzle pieces than just a keyboard, a display and a radio.

**Comment by srem1** [comment]
But you'd probably want a little better equipment than what's out there. I, for example, could not install a client program on my old phone. I would like to have everything in one. Something that I just run.

Another issue is batteries and power in general. 



So what would be the problem with making hardware like what I have now but adding a keyboard and say 2 battery slots. Different batteries. e.g. one AA the other 18650. Stepper-up-down would allow you to power it with whatever you happen to have from 0.3V to 5.5V 

You have one battery, good, it will work. In the same way, I would like it to be possible to set to use only one battery at a time or a second option of both at once. One AA its ok, 2 AA its ok too, 2 x 18650 its ok etc.

Tell me how much it would cost to add some kind of keyboard from a phone to RNode or some other keyboard with a minimum of 16 keys or a minimum of a wheel+10 keys

**Reply by markqvist** [reply]
> But you'd probably want a little better equipment than what's out there. I, for example, could not install a client program on my old phone. I would like to have everything in one. Something that I just run.

Yes, I completely understand this rationale. A lot of people want this.

All of the hardware stuff is easy. With the abundance of all kinds of components available, it is also possible to make it super cheap.

The only important thing right now is really the construction of a *system* and the *software* that makes it all work together, that makes it all possible. This is what Reticulum is about.

Then everyone can put together whatever kind of stuff they want to make these kind of devices, that can actually all interoperate with each other, and with programs running on computers, phones, servers, tablets, microcontrollers or anything really. This is the goal.

**Reply by markqvist** [reply]
Or to put it another way: The hardware is a useless if you don't have the right software and firmware to run it. Therefore I build software that can be used to transform all kinds of hardware into comms systems.

Right now you need a computer/phone/tablet to do the heavy lifting of running Reticulum. Soon this will not be necessary.

**Comment by srem1** [comment]
As of today, I can't verify that the reticulum works. I can't even figure out if 2 nodes with each other work. Min. because of the hardware. For me, the equipment is not easy. It is as difficult as software.

1. I asked how much it would cost to make similar hardware. And to publish the plans. This is where you can start and raise money. If more people gathered on tindie people would be happy to buy.

2. software as you know it is difficult to write for everything that exists. Therefore, reference hardware would be welcome. That is why I propose this idea. The author is open and does not want to write the system himself. Two people in the team.

3. treat this thread also as a discussion of what users want. Power supply is a very important issue. I would like it to be a powerbank and flashlight, for example.

**Reply by markqvist** [reply]
Ah, I think I didn't uderstand you totally correctly ;) I know, building stuff like this is *not* easy for most users. That is also why I try to focus my energy on developing open source plans that can be made quite easily by people who are relatively comfortable with hardware and electronics. Then they can give/sell this to other people that just want something in their hands that work. I think this is the best model. 

I was talking about the fact, that putting together a hardware system is easy, once you have a good foundation and system to build it on. I don't want to spend all my time soldering a thousand devices a month ;) I want to build the system that let's everyone build a thousand devices a month.

For me, building a device like the Armachat would be a "for fun" project. It is not where it is most important to spend the time and resources I have. It is much better that I write the systems, software and build plans.

> I asked how much it would cost to make similar hardware. And to publish the plans. This is where you can start and raise money. If more people gathered on tindie people would be happy to buy.

Do you mean, as in designing the hardware, publishing the plans, and then also selling the hardware? Or do you mean as in raising money from people, then design the thing and release it all open source?

> treat this thread also as a discussion of what users want. Power supply is a very important issue. I would like it to be a powerbank and flashlight, for example.

I hear you!

**Comment by fab1an2** [comment]
I can only speak for myself, but I would like an indiegogo or tindie campaign. A ready-made board is something that would make things easier for most normal user electronics.

Simple hat for popular Liligo or other esp32 with keyboard and power look like great.
